packages/firebase_database/example/android/app/src/main/java/io/flutter/plugins/firebasedatabaseexample/MainActivity.java

```diff
@@ -4,14 +4,23 @@
 
 package io.flutter.plugins.firebasedatabaseexample;
 
-import android.os.Bundle;
-import io.flutter.app.FlutterActivity;
-import io.flutter.plugins.GeneratedPluginRegistrant;
+import io.flutter.embedding.android.FlutterActivity;
+import io.flutter.embedding.engine.FlutterEngine;
+import io.flutter.embedding.engine.plugins.shim.ShimPluginRegistry;
+import io.flutter.plugins.firebase.core.FirebaseCorePlugin;
+import io.flutter.plugins.firebase.database.FirebaseDatabasePlugin;
 
 public class MainActivity extends FlutterActivity {
-  @Override
-  protected void onCreate(Bundle savedInstanceState) {
-    super.onCreate(savedInstanceState);
-    GeneratedPluginRegistrant.registerWith(this);
-  }
+
+    // TODO(cyanglaz): Remove this once v2 of GeneratedPluginRegistrant rolls to stable.
+    // https://github.com/flutter/flutter/issues/42694
+    @Override
+    public void configureFlutterEngine(FlutterEngine flutterEngine) {
+        super.configureFlutterEngine(flutterEngine);
+        flutterEngine.getPlugins().add(new FirebaseDatabasePlugin());
+
+        ShimPluginRegistry shimPluginRegistry = new ShimPluginRegistry(flutterEngine);
```

Comment:
This is the new embedding example right? Looks like firebase core was already migrated. I don't think you need this shim?